{"text": "2024-02-06 19:21:48 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:03.694313", "seconds": 3.694313}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:21:48.596680+08:00", "timestamp": 1707218508.59668}}}
{"text": "2024-02-06 19:21:57 | INFO    | __main__:claim_epic_games:97 | Try to flush cookie - {'task': 'claim_epic_games'}\n", "record": {"elapsed": {"repr": "0:00:12.979143", "seconds": 12.979143}, "exception": null, "extra": {"task": "claim_epic_games"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "claim_epic_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 97, "message": "Try to flush cookie", "module": "claim", "name": "__main__", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:21:57.881510+08:00", "timestamp": 1707218517.88151}}}
{"text": "2024-02-06 19:21:59 | SUCCESS    | epic_games.agent:_login:253 | login - {'result': 'Successfully refreshed tokens'}\n", "record": {"elapsed": {"repr": "0:00:14.728165", "seconds": 14.728165}, "exception": null, "extra": {"result": "Successfully refreshed tokens"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "_login", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 253, "message": "login", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:21:59.630532+08:00", "timestamp": 1707218519.630532}}}
{"text": "2024-02-06 19:22:06 | SUCCESS    | epic_games.agent:flush_token:280 | flush_token - {'path': PosixPath('/home/runner/work/epic/epic/epic/user_data_dir/epic-games@epicgames/ctx_cookie.json')}\n", "record": {"elapsed": {"repr": "0:00:21.694928", "seconds": 21.694928}, "exception": null, "extra": {"path": "/home/runner/work/epic/epic/epic/user_data_dir/epic-games@epicgames/ctx_cookie.json"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "flush_token", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 280, "message": "flush_token", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:22:06.597295+08:00", "timestamp": 1707218526.597295}}}
{"text": "2024-02-06 19:22:06 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:21.776612", "seconds": 21.776612}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:22:06.678979+08:00", "timestamp": 1707218526.678979}}}
{"text": "2024-02-06 19:22:06 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Doors - Paradox', 'url': 'https://store.epicgames.com/en-US/p/doors-paradox-5896c7'}\n", "record": {"elapsed": {"repr": "0:00:22.034075", "seconds": 22.034075}, "exception": null, "extra": {"title": "Doors - Paradox", "url": "https://store.epicgames.com/en-US/p/doors-paradox-5896c7"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:22:06.936442+08:00", "timestamp": 1707218526.936442}}}
{"text": "2024-02-06 19:22:06 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/doors-paradox-5896c7'}\n", "record": {"elapsed": {"repr": "0:00:22.034681", "seconds": 22.034681}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/doors-paradox-5896c7"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:22:06.937048+08:00", "timestamp": 1707218526.937048}}}
{"text": "2024-02-06 19:22:37 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/doors-paradox-5896c7'}\n", "record": {"elapsed": {"repr": "0:00:52.537346", "seconds": 52.537346}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/doors-paradox-5896c7"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:22:37.439713+08:00", "timestamp": 1707218557.439713}}}
{"text": "2024-02-06 19:23:07 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/doors-paradox-5896c7'}\n", "record": {"elapsed": {"repr": "0:01:23.039644", "seconds": 83.039644}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/doors-paradox-5896c7"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic/epic/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:23:07.942011+08:00", "timestamp": 1707218587.942011}}}
{"text": "2024-02-06 19:23:38 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3623), thread 'MainThread' (140544120400768): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 88, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 47, in __call__\n    do = self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 325, in iter\n    raise retry_exc.reraise()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 158, in reraise\n    raise self.last_attempt.result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 50, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/epic/epic/epic/src/epic_games/agent.py\", line 295, in claim_weekly_games\n    await page.goto(promotion.url, wait_until=\"load\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 9275, in goto\n    await self._impl_obj.goto(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_page.py\", line 484, in goto\n    return await self._main_frame.goto(**locals_to_params(locals()))\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_frame.py\", line 149, in goto\n    await self._channel.send(\"goto\", locals_to_params(locals()))\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 63, in send\n    return await self._connection.wrap_api_call(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 495, in wrap_api_call\n    return await cb()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_connection.py\", line 101, in inner_send\n    result = next(iter(done)).result()\nplaywright._impl._errors.TimeoutError: Timeout 30000ms exceeded.\n", "record": {"elapsed": {"repr": "0:01:53.225338", "seconds": 113.225338}, "exception": {"type": "TimeoutError", "value": "Timeout 30000ms exceeded.", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3623), thread 'MainThread' (140544120400768):", "module": "claim", "name": "__main__", "process": {"id": 3623, "name": "MainProcess"}, "thread": {"id": 140544120400768, "name": "MainThread"}, "time": {"repr": "2024-02-06 19:23:38.127705+08:00", "timestamp": 1707218618.127705}}}
